src/Symfony/Component/Process/Process.php

```diff
@@ -1632,6 +1635,19 @@ private function escapeArgument(?string $argument): string
         return '"'.str_replace(['"', '^', '%', '!', "\n"], ['""', '"^^"', '"^%"', '"^!"', '!LF!'], $argument).'"';
     }
 
+    private function replacePlaceholders(string $commandline, array $env)
+    {
+        return preg_replace_callback('/"\$([_a-zA-Z]++[_a-zA-Z0-9]*+)"/', function ($matches) use ($commandline, $env) {
+            $m = $matches[1];
+
+            if (!isset($env[$m]) || (isset($env[$m]) && false === $env[$m])) {
+                throw new InvalidArgumentException(sprintf('Invalid command line "%s": no values provided for any placeholders.', $commandline));
```

Comment:
the message should tell the missing key.
`Command line is missing a value for key "$FOO": [...]`